src/main/java/edu/hawaii/its/api/service/UpdateMemberService.java

```diff
@@ -318,83 +319,93 @@ public void checkIfSelfOptOrAdmin(String currentUser, String identifier) {
 
     private GroupingMoveMembersResult moveGroupMembers(String currentUser, String addGroupPath, String removeGroupPath,
             List<String> uhIdentifiers) {
-        RemoveMembersResults removeMembersResults = grouperApiService.removeMembers(currentUser, removeGroupPath, uhIdentifiers);
-        AddMembersResults addMembersResults = grouperApiService.addMembers(currentUser, addGroupPath, uhIdentifiers);
+        RemoveMembersResults removeMembersResults = grouperService.removeMembers(currentUser, removeGroupPath, uhIdentifiers);
+        AddMembersResults addMembersResults = grouperService.addMembers(currentUser, addGroupPath, uhIdentifiers);
         GroupingMoveMembersResult result = new GroupingMoveMembersResult(addMembersResults, removeMembersResults);
-        timestampService.update(result.getAddResults());
-        timestampService.update(result.getRemoveResults());
+        if(grouperService instanceof GrouperApiOOTBService) {
+            timestampService.update(result.getAddResults());
+            timestampService.update(result.getRemoveResults());
+        }
```

Comment:
For all the private functions in this service, we still need to update the timestamp using the real Grouper.